core/src/Pos/Core/Genesis/Canonical.hs

```diff
@@ -8,54 +8,18 @@ module Pos.Core.Genesis.Canonical
-import           Data.Time.Units (Millisecond, Second, convertUnit)
-import           Data.Typeable (typeRep)
-import           Formatting (formatToString)
+import           Data.Time.Units (Millisecond)
 import qualified Formatting.Buildable as Buildable
 import           Serokell.Data.Memory.Units (Byte)
-import           Serokell.Util.Base16 (base16F)
-import qualified Serokell.Util.Base16 as B16
-import           Serokell.Util.Base64 (base64F)
-import qualified Serokell.Util.Base64 as B64
 import           Serokell.Util.Text (readDecimal, readUnsignedDecimal)
-import           Text.JSON.Canonical (FromJSON (..), FromObjectKey (..), Int54,
+import           Text.JSON.Canonical (FromJSON (..), FromObjectKey (..),
                      JSValue (..), ReportSchemaErrors (expected), ToJSON (..),
-                     ToObjectKey (..), expectedButGotValue, fromJSField,
-                     fromJSObject, mkObject)
+                     ToObjectKey (..), expectedButGotValue, fromJSObject)
 
-import           Pos.Binary.Class (AsBinary (..))
-import           Pos.Core.Common (Address, Coeff (..), Coin (..),
-                     CoinPortion (..), SharedSeed (..), StakeholderId,
-                     TxFeePolicy (..), TxSizeLinear (..), addressF,
-                     decodeTextAddress, getCoinPortion, unsafeGetCoin)
-import           Pos.Core.Delegation (HeavyDlgIndex (..), ProxySKHeavy)
-import           Pos.Core.ProtocolConstants (VssMaxTTL (..), VssMinTTL (..))
-import           Pos.Core.Slotting (EpochIndex (..), Timestamp (..))
-import           Pos.Core.Ssc (VssCertificate (..), VssCertificatesMap (..),
-                     validateVssCertificatesMap)
-import           Pos.Core.Update (BlockVersionData (..), SoftforkRule (..))
-import           Pos.Crypto (ProxyCert, ProxySecretKey (..), PublicKey,
-                     RedeemPublicKey, Signature, decodeAbstractHash,
-                     fromAvvmPk, fullProxyCertHexF, fullPublicKeyF,
-                     fullSignatureHexF, hashHexF, parseFullProxyCert,
-                     parseFullPublicKey, parseFullSignature, redeemPkB64UrlF)
-import           Pos.Crypto.Configuration (ProtocolMagic (..))
-
-import           Pos.Core.Genesis.AvvmBalances (GenesisAvvmBalances (..))
-import           Pos.Core.Genesis.Data (GenesisData (..))
-import           Pos.Core.Genesis.Delegation (GenesisDelegation (..),
-                     recreateGenesisDelegation)
-import           Pos.Core.Genesis.NonAvvmBalances (GenesisNonAvvmBalances (..))
-import           Pos.Core.Genesis.ProtocolConstants
-                     (GenesisProtocolConstants (..))
-import           Pos.Core.Genesis.VssCertificatesMap
-                     (GenesisVssCertificatesMap (..))
-import           Pos.Core.Genesis.WStakeholders (GenesisWStakeholders (..))
+import           Pos.Util.Json.Parse (tryParseString)
 
 ----------------------------------------------------------------------------
```

Comment:
Some of the instances from this module will have to remain as orphans so I've gone ahead and left them in here.